tasks/sanity-inspect-image.yaml

```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sanity-inspect-image
+spec:
+  description: >-
+    Get manifest data for the source image and its base image to workspace
+  params:
+    - name: IMAGE_URL
+      description: the fully qualified image name
+      type: string
+  results:
+    - description: Base image the source image is built from
+      name: BASE_IMAGE
+  steps:
+  - name: inspect-image
+    image: quay.io/redhat-appstudio/hacbs-test:latest
+    script: |
+      IMAGE_INSPECT=/tmp/image_inspect.json
+      BASE_IMAGE_INSPECT=/tmp/base_image_inspect.json
+      RAW_IMAGE_INSPECT=/tmp/raw_image_inspect.json
+
+      echo "Inspecting manifest for source image $(params.IMAGE_URL)"
+      skopeo inspect --no-tags docker://$(params.IMAGE_URL) > $IMAGE_INSPECT
+      skopeo inspect --no-tags --raw docker://$(params.IMAGE_URL) > $RAW_IMAGE_INSPECT
+
+      echo "Getting base image manifest for source image $(params.IMAGE_URL)"
+      BASE_IMAGE_NAME="$(jq -r ".annotations.\"org.opencontainers.image.base.name\"" $RAW_IMAGE_INSPECT)"
+      BASE_IMAGE_DIGEST="$(jq -r ".annotations.\"org.opencontainers.image.base.digest\"" $RAW_IMAGE_INSPECT)"
+      if [ $BASE_IMAGE_NAME == 'null' ]; then
+        echo "Cannot get base image info from 'annotations'"
+        echo "Trying to get base image info from 'Labels'"
+        BASE_IMAGE_NAME="$(jq -r ".Labels.\"org.opencontainers.image.base.name\"" $IMAGE_INSPECT)"
+        BASE_IMAGE_DIGEST="$(jq -r ".annotations.\"org.opencontainers.image.base.digest\"" $IMAGE_INSPECT)"
+        if [ "$BASE_IMAGE_NAME" == 'null' ]; then
+          echo "Cannot get base image info from 'Labels', please check the source image $(params.IMAGE_URL)"
+          exit 0
+        fi
+      fi
+      if [ -z "$BASE_IMAGE_NAME" ]; then
+        echo "Source image $(params.IMAGE_URL) is built from scratch, so there is no base image"
+        exit 0
+      fi
+      BASE_IMAGE="${BASE_IMAGE_NAME%:*}@$BASE_IMAGE_DIGEST"
+      echo "The base image is $BASE_IMAGE, get its manifest now"
+      skopeo inspect --no-tags docker://$BASE_IMAGE  > $BASE_IMAGE_INSPECT
```

Comment:
When running via s2i builder it's failing since it's trying to inspect internal registry:
```
+ skopeo inspect --no-tags docker://image-registry.openshift-image-registry.svc:5000/openshift/nodejs@sha256:1451b4849c535edfc238b11e85d8bbc9599516fb9d079bb8e116051b532a04a0
time="2022-04-21T08:50:26Z" level=fatal msg="Error parsing image name \"docker://image-registry.openshift-image-registry.svc:5000/openshift/nodejs@sha256:1451b4849c535edfc238b11e85d8bbc9599516fb9d079bb8e116051b532a04a0\": reading manifest sha256:1451b4849c535edfc238b11e85d8bbc9599516fb9d079bb8e116051b532a04a0 in image-registry.openshift-image-registry.svc:5000/openshift/nodejs: manifest unknown: manifest unknown"
```

Probably it could be enough to ignore this right now -> `skopeo inspect --no-tags docker://$BASE_IMAGE  > $BASE_IMAGE_INSPECT || true`

Reply:
Ok, thanks. Let use it for now. Updated.